lib/hammer_cli_foreman/commands.rb

```diff
@@ -610,8 +608,7 @@ def associated_resource
 
     def self.associated_resource(name=nil)
       @associated_api_resource = HammerCLIForeman.foreman_resource!(name) unless name.nil?
-      return @associated_api_resource if @associated_api_resource
-      return superclass.associated_resource if superclass.respond_to? :associated_resource
+      return (@associated_api_resource ||= nil) || (superclass.associated_resource if superclass.respond_to? :associated_resource)
```

Comment:
I think this is getting too complex. Isn't it better to split it over multiple lines? Though I'll admit I can't figure out what is a pretty format. The whole pattern of combining a getter and setter is tricky.